x-pack/plugins/observability_solution/apm/public/components/shared/try_it_button/index.tsx

```diff
@@ -0,0 +1,193 @@
+          })}
+        />
+      </EuiFlexItem>
+    );
+  }
+
+  function Icon() {
+    return (
+      <>
+        {icon && (
+          <EuiFlexItem grow={false}>
+            <EuiBetaBadge
+              color="hollow"
+              iconType={icon}
+              label={
+                icon === 'beaker'
+                  ? i18n.translate('xpack.apm.tryIt.techPreview', {
+                      defaultMessage: 'Technical preview',
+                    })
+                  : i18n.translate('xpack.apm.tryIt.beta', {
+                      defaultMessage: 'Beta',
+                    })
+              }
+            />
+          </EuiFlexItem>
+        )}
+      </>
+    );
+  }
+
+  function PromoLabel() {
+    return (
+      <>
+        {!!promoLabel && (
+          <EuiFlexItem grow={false}>
+            <EuiText>{promoLabel}</EuiText>
+          </EuiFlexItem>
+        )}
+      </>
+    );
+  }
+
+  function Popover() {
+    return (
+      <>
+        {popoverContent || !canEditAdvancedSettings ? (
+          <EuiFlexItem grow={false}>
+            <EuiPopover
+              button={
+                <EuiButtonIcon
+                  data-test-subj="apmPopoverButton"
+                  iconType="questionInCircle"
+                  aria-label={i18n.translate(
+                    'xpack.apm.tryItButton.euiButtonIcon.tryItHelperButtonLabel',
+                    { defaultMessage: 'Try it helper button' }
+                  )}
+                  onClick={togglePopover}
+                />
+              }
+              isOpen={isPopoverOpen}
+              closePopover={togglePopover}
+              anchorPosition="upCenter"
+            >
+              <>
+                {popoverContent}
+                {!canEditAdvancedSettings && (
+                  <>
+                    <EuiSpacer size="s" />
+                    {i18n.translate(
+                      'xpack.apm.tryItButton.euiButtonIcon.adminAccess',
+                      {
+                        defaultMessage:
+                          'Please ask your administrator to turn it on by enabling it in within settings.',
+                      }
+                    )}
+                  </>
+                )}
+              </>
+            </EuiPopover>
+          </EuiFlexItem>
+        ) : null}
+      </>
+    );
+  }
+
+  function Link() {
+    return (
+      <>
+        {linkLabel && canEditAdvancedSettings && (
+          <EuiFlexItem grow={false}>
+            <EuiLink
+              data-test-subj="apmLink"
+              disabled={isLoading}
+              onClick={onClick}
+            >
+              {linkLabel}
+            </EuiLink>
+          </EuiFlexItem>
+        )}
+      </>
+    );
+  }
+
+  function Loading() {
+    return (
+      <>
+        {isLoading && (
+          <EuiFlexItem grow={false}>
+            <EuiLoadingSpinner size="s" />
+          </EuiFlexItem>
+        )}
+      </>
+    );
+  }
+
+  if (isFeatureEnabled) {
+    return (
```

Comment:
Since the wrapping `EuiFlexGroup` and all the `children` have identical props in either case, we could instead pass `isFeatureEnabled` as a prop to `TryItBadge` and `PromoLabel`, and have them return `null` when `isFeatureEnabled === true`. This lets us simplify this final `return` logic. Something like:

```typescript
  return (
    <EuiFlexGroup gutterSize="s" alignItems="center">
      <TryItBadge isFeatureEnabled={isFeatureEnabled} />
      <Icon />
      <PromoLabel isFeatureEnabled={isFeatureEnabled} />
      <Popover />
      <Link />
      <Loading />
    </EuiFlexGroup>
  );
```

Reply:
I didn't do that because if you look closely, when `isFeatureEnabled = true` we invert the order of the `Link` and `Popover` components, so I think this way looks neat and clean.